power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,114 @@
+            ComplexTensor4 y = r_matrix + 1.0i * x_matrix;
+            z_series_abc = kron_reduction(y);
+        }
+        return z_series_abc;
+    }
+
+    ComplexTensor<asymmetric_t> compute_c_matrix_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> c_matrix;
+        if (!is_nan(asym_line_input.c0) && !is_nan(asym_line_input.c1)) {
+            c_matrix = ComplexTensor<asymmetric_t>{(2.0 * asym_line_input.c1 + asym_line_input.c0) / 3.0, (asym_line_input.c0 - asym_line_input.c1) / 3.0 };
+        } 
+        else if (is_nan(asym_line_input.c_nn)) {
+            c_matrix = ComplexTensor<asymmetric_t>(asym_line_input.c_aa, asym_line_input.c_bb, asym_line_input.c_cc, asym_line_input.c_ba, asym_line_input.c_ca, asym_line_input.c_cb);
+        } 
+        else {
+            ComplexTensor4 c_matrix_neutral = ComplexTensor4(asym_line_input.c_aa, asym_line_input.c_bb, asym_line_input.c_cc, asym_line_input.c_nn, asym_line_input.c_ba, asym_line_input.c_ca, asym_line_input.c_na, asym_line_input.c_cb, asym_line_input.c_nb, asym_line_input.c_nc);
+            c_matrix = kron_reduction(c_matrix_neutral);
+        }
+        return c_matrix;
+    }
+
+    BranchCalcParam<symmetric_t> sym_calc_param() const override {
+        DoubleComplex y1_series_ = average_of_diagonal_of_matrix(_y_series_abc) - average_of_off_diagonal_of_matrix(_y_series_abc);
+        DoubleComplex y1_shunt_ = average_of_diagonal_of_matrix(_y_shunt_abc) - average_of_off_diagonal_of_matrix(_y_shunt_abc);
+        return calc_param_y_sym(y1_series_, y1_shunt_, 1.0);
+    }
+
+    BranchCalcParam<asymmetric_t> asym_calc_param() const override {
+        BranchCalcParam<asymmetric_t> param{};
+        // not both connected
+        if (!branch_status()) {
+            // single connected
+            if (from_status() || to_status()) {
+                // branch_shunt = 0.5 * y_shunt + 1.0 / (1.0 / y_series + 2.0 / y_shunt);
+                ComplexTensor<asymmetric_t> branch_shunt = ComplexTensor<asymmetric_t>();
+                if ((cabs(_y_shunt_abc) >= numerical_tolerance).all()) {
+                    branch_shunt = 0.5 * inv(_y_shunt_abc) + inv(inv(_y_series_abc) + 2.0 * inv(_y_shunt_abc));
+                }
+                // from or to connected
+                param.yff() = from_status() ? branch_shunt : ComplexTensor<asymmetric_t>();
+                param.ytt() = to_status() ? branch_shunt :  ComplexTensor<asymmetric_t>();
+            }
+        }
+        // both connected
+        else {
+            param.ytt() = _y_series_abc + 0.5 * _y_shunt_abc;
+            param.yff() = param.ytt();
+            param.yft() = _y_series_abc;
+            param.ytf() = _y_series_abc;
```

Comment:
There should be a minus sign here.

Reply:
See
https://github.com/PowerGridModel/power-grid-model/blob/24eaaf9adffe058219fe44273f0ee8c28db599dd/power_grid_model_c/power_grid_model/include/power_grid_model/component/branch.hpp#L228-L233